src/averageAges.js

```diff
@@ -52,7 +76,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let childrenList = people.filter(function hasMotherInList({ mother }) {
+    return people.some(({ name }) => mother === name);
+  });
+
+  childrenList = (onlyWithSon === true)
+    ? childrenList.filter(({ sex }) => sex === 'm')
+    : childrenList;
+
+  const ageDiffList = childrenList.map(function findAgeDiff({ mother, born }) {
+    const motherRef = people.find(({ name }) => name === mother);
+
+    return born - motherRef.born;
+  });
```

Comment:
There's no need to create another function inside 'map' method (especially if you don't use it never again) because there's already a callback-function there. You can just use 'find' method inside your 'map' (in the way you've done it already). E. g.
```
 const ageDiffList = childrenList.map(child =>
    child.born - (people.find(mother => mother.name === child.mother)).born
)
```